crates/meroctl/src/cli/context/create.rs

```diff
@@ -0,0 +1,177 @@
+}
+
+#[derive(Debug, Parser)]
+#[group(requires_all(&["dev", "path", "version"]))]
+pub struct CreateCommand {
+    /// The application ID to attach to the context
+    #[clap(long, short = 'a', exclusive = true, value_name = "APP_ID")]
+    application_id: String,
+
+    /// Enable dev mode
+    #[clap(long, group = "dev_args")]
+    dev: bool,
+
+    /// Path to use in dev mode
+    #[clap(short, long, group = "dev_args", value_name = "PATH")]
+    path: Utf8PathBuf,
+
+    /// Version of the application
+    #[clap(
+        short,
+        long,
+        group = "dev_args",
+        default_value = "0.0.0",
+        value_name = "VERSION"
+    )]
+    version: Version,
+}
+
+impl CreateCommand {
+    pub async fn run(self, root_args: RootArgs) -> eyre::Result<()> {
+        let path = root_args.home.join(&root_args.node_name);
+
+        if !ConfigFile::exists(&path) {
+            eyre::bail!("Config file does not exist")
+        };
+
+        let Ok(config) = ConfigFile::load(&path) else {
+            eyre::bail!("Failed to load config file")
+        };
+
+        let Some(multiaddr) = config.network.server.listen.first() else {
+            eyre::bail!("No address.")
+        };
+
+        if self.dev {
+            return link_local_app(multiaddr, self.path, self.version).await;
+        }
+        create_context(multiaddr, self.application_id).await
+    }
+}
+
+async fn create_context(base_multiaddr: &Multiaddr, application_id: String) -> eyre::Result<()> {
+    app_installed(&base_multiaddr, &application_id).await?;
+
+    let url = multiaddr_to_url(base_multiaddr, "admin-api/contexts-dev")?;
+    let client = Client::new();
+    let request = CreateContextRequest { application_id };
+
+    let response = client.post(url).json(&request).send().await?;
+
+    if response.status().is_success() {
+        let context_response: CreateContextResponse = response.json().await?;
+        let context = context_response.data;
+
+        println!("Context created successfully:");
+        println!("ID: {}", context.id);
+        println!("Application ID: {}", context.application_id);
+    } else {
+        let status = response.status();
+        let error_text = response.text().await?;
+        eyre::bail!(
+            "Request failed with status: {}. Error: {}",
+            status,
+            error_text
+        );
+    }
+    Ok(())
+}
+
+async fn app_installed(base_multiaddr: &Multiaddr, application_id: &String) -> eyre::Result<()> {
+    let url = multiaddr_to_url(base_multiaddr, "admin-api/applications-dev")?;
+    let client = Client::new();
+    let response = client.get(url).send().await?;
+    if response.status().is_success() {
+        let api_response: ListApplicationsResponse = response.json().await?;
+        let app_list = api_response.data.apps;
+        let x = app_list
+            .iter()
+            .any(|app| app.id.as_ref() == *application_id);
+        if x {
+            return Ok(());
```

Comment:
looks like all paths in this function short-circuit so it never gets to the end, refactor this so `Ok(())` ends up at the bottom of the function